src/components/message/MessagesAttachedSelection.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { observer } from 'mobx-react-lite';
+import { useMessagesWorkspaceStore } from '../../hooks';
+import { EventMessage } from '../../models/EventMessage';
+
+const MessageAttachedSelection = () => {
+	const messagesStore = useMessagesWorkspaceStore();
+
+	const { attachedMessages } = messagesStore;
+
+	const [currentAttachedMessages, setCurrentAttachedMessages] = React.useState<EventMessage[]>([]);
+	const [messageIndex, setMessageIndex] = React.useState<number>(0);
+
+	React.useEffect(() => {
+		if (attachedMessages !== currentAttachedMessages) {
```

Comment:
i think we dont even need to compare it, hook will be called every time we load new attached messages so we can delete state `currentAttachedMessages`